src/tree.rs

```diff
@@ -372,6 +416,18 @@ impl<C: TreeConfig> SyntaxTree<C> {
         (kind, start, end)
     }
 
+    // It seems that the three unsafe functions can be replaced with one safe:
+    // ```
+    // enum EventKind { Start, Add, Finish }
+    //
+    // fn get_kind(&self, idx: u32) -> Option<EventKind>
+    // ```
+    //
+    // The call-site is doing the range check anyway, so we can fold that into
+    // an Option.
```

Comment:
Not all call sites are explicitly doing bounds checks at the moment, though, such as those in the `SyntaxNode` iterators. That would incur an `unwrap()` there, which I think would hurt performance (and would make the code messier).